### PR TITLE
Core/Porting OMP Parallel for loops to Parallel Utilities - 14

### PR DESCRIPTION
**Pull request by ashishdarekar**
**Description**
This PR ports the OMP Parallel Loops to Parallel Utilities
 
- ParallelUtilities

**Changelog**
-Replaced Simple OMP For loop with Parallel Utilities - IndexPartition()
-Replaced OMP For loops with Private variables with Parallel Utilities - IndexPartition()
-Replaced OMP Parallel For loops with Private variables with ParallelUtilities(TLS)
-Still few loops from the same file need to port after clarification from PR #8803

### CODE REVIEW

**Comment by loumalouomega** [comment]
Fails in CentOS

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 1]
```diff
@@ -593,12 +593,11 @@ class ResidualBasedBlockBuilderAndSolver
         // The goal is that the stiffness is computed with the
         // converged configuration at the end of the previous step.
         const auto it_dof_begin = BaseType::mDofSet.begin();
-        #pragma omp parallel for
-        for (int i = 0; i < static_cast<int>(BaseType::mDofSet.size()); ++i) {
-            auto it_dof = it_dof_begin + i;
+        IndexPartition<std::size_t>(static_cast<int>(BaseType::mDofSet.size())).for_each([&](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: I think you can use the `block_for_each` here

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 2]
```diff
@@ -699,15 +698,13 @@ class ResidualBasedBlockBuilderAndSolver
         const int ndofs = static_cast<int>(BaseType::mDofSet.size());
 
         //NOTE: dofs are assumed to be numbered consecutively in the BlockBuilderAndSolver
-        #pragma omp parallel for firstprivate(ndofs)
-        for (int k = 0; k<ndofs; k++)
-        {
-            typename DofsArrayType::iterator dof_iterator = BaseType::mDofSet.begin() + k;
+        IndexPartition<std::size_t>(ndofs).for_each([&](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: same

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 3]
```diff
@@ -853,12 +850,10 @@ class ResidualBasedBlockBuilderAndSolver
         BaseType::mEquationSystemSize = BaseType::mDofSet.size();
         int ndofs = static_cast<int>(BaseType::mDofSet.size());
 
-        #pragma omp parallel for firstprivate(ndofs)
-        for (int i = 0; i < static_cast<int>(ndofs); i++) {
-            typename DofsArrayType::iterator dof_iterator = BaseType::mDofSet.begin() + i;
-            dof_iterator->SetEquationId(i);
-
-        }
+        IndexPartition<std::size_t>(ndofs).for_each([&](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: same

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 4]
```diff
@@ -991,13 +986,12 @@ class ResidualBasedBlockBuilderAndSolver
         const int ndofs = static_cast<int>(BaseType::mDofSet.size());
 
         //NOTE: dofs are assumed to be numbered consecutively in the BlockBuilderAndSolver
-        #pragma omp parallel for firstprivate(ndofs)
-        for (int k = 0; k<ndofs; k++) {
-            typename DofsArrayType::iterator dof_iterator = BaseType::mDofSet.begin() + k;
+        IndexPartition<std::size_t>(ndofs).for_each([&](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: same

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 5]
```diff
@@ -1068,26 +1061,25 @@ class ResidualBasedBlockBuilderAndSolver
             }
         }
 
-        #pragma omp parallel for firstprivate(system_size)
-        for (int k = 0; k < static_cast<int>(system_size); ++k) {
-            std::size_t col_begin = Arow_indices[k];
-            std::size_t col_end = Arow_indices[k+1];
-            const double k_factor = scaling_factors[k];
+        IndexPartition<int >(static_cast<int>(system_size)).for_each([&](int Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: please use `std::size_t`
`int` was only used bcs of OpenMP

This applies generally

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 6]
```diff
@@ -1116,13 +1108,12 @@ class ResidualBasedBlockBuilderAndSolver
             TSparseSpace::Copy(b_modified, rb);
 
             // Apply diagonal values on slaves
-            #pragma omp parallel for
-            for (int i = 0; i < static_cast<int>(mSlaveIds.size()); ++i) {
-                const IndexType slave_equation_id = mSlaveIds[i];
+            IndexPartition<std::size_t>(static_cast<int>(mSlaveIds.size())).for_each([&](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: same

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 7]
```diff
@@ -1165,14 +1156,13 @@ class ResidualBasedBlockBuilderAndSolver
             const double max_diag = GetMaxDiagonal(rA);
 
             // Apply diagonal values on slaves
-            #pragma omp parallel for
-            for (int i = 0; i < static_cast<int>(mSlaveIds.size()); ++i) {
-                const IndexType slave_equation_id = mSlaveIds[i];
+            IndexPartition<std::size_t>(static_cast<int>(mSlaveIds.size())).for_each([&](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: same

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 8]
```diff
@@ -1452,21 +1442,20 @@ class ResidualBasedBlockBuilderAndSolver
             for (int i = 0; i < static_cast<int>(mT.size1()); i++)
                 Trow_indices[i + 1] = Trow_indices[i] + indices[i].size();
 
-            #pragma omp parallel for
-            for (int i = 0; i < static_cast<int>(mT.size1()); ++i) {
-                const IndexType row_begin = Trow_indices[i];
-                const IndexType row_end = Trow_indices[i + 1];
+            IndexPartition<std::size_t>(static_cast<int>(mT.size1())).for_each([&](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: same

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 9]
```diff
@@ -1579,63 +1568,66 @@ class ResidualBasedBlockBuilderAndSolver
         std::vector< LockObject > lock_array(equation_size);
 
         std::vector<std::unordered_set<std::size_t> > indices(equation_size);
-        #pragma omp parallel for firstprivate(equation_size)
-        for (int iii = 0; iii < static_cast<int>(equation_size); iii++) {
-            indices[iii].reserve(40);
-        }
+
+        IndexPartition<std::size_t>(equation_size).for_each([&indices](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: `block_for_each`

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 10]
```diff
@@ -1579,63 +1568,66 @@ class ResidualBasedBlockBuilderAndSolver
...
-        #pragma omp parallel for firstprivate(nelements, ids)
-        for (int iii=0; iii<nelements; iii++) {
-            typename ElementsContainerType::iterator i_element = el_begin + iii;
-            pScheme->EquationId(*i_element, ids, CurrentProcessInfo);
-            for (std::size_t i = 0; i < ids.size(); i++) {
-                lock_array[ids[i]].lock();
-                auto& row_indices = indices[ids[i]];
-                row_indices.insert(ids.begin(), ids.end());
-                lock_array[ids[i]].unlock();
+        IndexPartition<std::size_t>(nelements).for_each(ids, [&](std::size_t Index, Element::EquationIdVectorType& rIds){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: same

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 11]
```diff
@@ -1579,63 +1568,66 @@ class ResidualBasedBlockBuilderAndSolver
...
-            typename ConditionsArrayType::iterator i_condition = cond_begin + iii;
-            pScheme->EquationId(*i_condition, ids, CurrentProcessInfo);
-            for (std::size_t i = 0; i < ids.size(); i++) {
-                lock_array[ids[i]].lock();
-                auto& row_indices = indices[ids[i]];
-                row_indices.insert(ids.begin(), ids.end());
-                lock_array[ids[i]].unlock();
+        });
+
+        IndexPartition<std::size_t>(nconditions).for_each(ids, [&](std::size_t Index, Element::EquationIdVectorType& rIds){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: same

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 12]
```diff
@@ -1579,63 +1568,66 @@ class ResidualBasedBlockBuilderAndSolver
...
+            tls.master_ids.resize(3,0);
 
             const int nmasterSlaveConstraints = rModelPart.MasterSlaveConstraints().size();
             const auto const_begin = rModelPart.MasterSlaveConstraints().begin();
 
-            #pragma omp parallel for firstprivate(nmasterSlaveConstraints, slave_ids, master_ids)
-            for (int iii = 0; iii<nmasterSlaveConstraints; ++iii) {
-                auto i_const = const_begin + iii;
-                i_const->EquationIdVector(slave_ids, master_ids, CurrentProcessInfo);
+            IndexPartition<std::size_t>(nmasterSlaveConstraints).for_each(tls, [&](std::size_t Index, TLS& rTls){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: same

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 13]
```diff
@@ -1659,22 +1651,21 @@ class ResidualBasedBlockBuilderAndSolver
             Arow_indices[i+1] = Arow_indices[i] + indices[i].size();
         }
 
-        #pragma omp parallel for
-        for (int i = 0; i < static_cast<int>(A.size1()); i++) {
-            const unsigned int row_begin = Arow_indices[i];
-            const unsigned int row_end = Arow_indices[i+1];
+        IndexPartition<int>(static_cast<int>(A.size1())).for_each([&](int Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: std::size_t

**Review by philbucher — COMMENTED** [review]
@ashishdarekar please see my comments

Lets see how the CentOS build behaves after the changes

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h (hunk 5 above)
Comment (ashishdarekar) [reply kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: Yes, but it was giving an error on line 1071 `if (static_cast<int>(Acol_indices[j]) != Index )`. 
Will remove that Typecasted `Acol_indices[j]` and will change accordingly. 
Thanks :)

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h (hunk 3 above)
Comment (ashishdarekar) [reply kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: @philbucher 
On line 849 `dof_iterator->SetEquationId(Index)`  how can I get Index ??
Is it possible to use block_for_each()?

Comment (ashishdarekar) [reply kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: I think we cant use block_for_each() here, Hence keeping IndexPartition()

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 14]
```diff
@@ -1849,13 +1813,13 @@ class ResidualBasedBlockBuilderAndSolver
         // Creating a buffer for parallel vector fill
         const int num_threads = ParallelUtilities::GetNumThreads();
         Vector max_vector(num_threads, 0.0);
-        #pragma omp parallel for
-        for(int i = 0; i < static_cast<int>(TSparseSpace::Size1(rA)); ++i) {
+
+        IndexPartition<std::size_t>(TSparseSpace::Size1(rA)).for_each([&](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: can you please revert the changes in this function?
I would like to do this properly with the MaxReduction, but in a different PR

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 15]
```diff
@@ -1882,13 +1846,13 @@ class ResidualBasedBlockBuilderAndSolver
         // Creating a buffer for parallel vector fill
         const int num_threads = ParallelUtilities::GetNumThreads();
         Vector min_vector(num_threads, std::numeric_limits<double>::max());
-        #pragma omp parallel for
-        for(int i = 0; i < static_cast<int>(TSparseSpace::Size1(rA)); ++i) {
+
+        IndexPartition<std::size_t>(TSparseSpace::Size1(rA)).for_each([&](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: same

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 16]
```diff
@@ -1814,10 +1778,10 @@ class ResidualBasedBlockBuilderAndSolver
     double GetDiagonalNorm(TSystemMatrixType& rA)
     {
         double diagonal_norm = 0.0;
-        #pragma omp parallel for reduction(+:diagonal_norm)
-        for(int i = 0; i < static_cast<int>(TSparseSpace::Size1(rA)); ++i) {
-            diagonal_norm += std::pow(rA(i,i), 2);
-        }
+        diagonal_norm = IndexPartition<std::size_t>(TSparseSpace::Size1(rA)).for_each<SumReduction<double>>([&](std::size_t Index){
+            return std::pow(rA(Index,Index), 2);
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h:1780]: unrelated to this PR:
@loumalouomega @RiccardoRossi can we be sure that this entry exists?
I think it should, but do we know for sure? It is a sparse matrix after all

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 17]
```diff
@@ -1659,22 +1624,21 @@ class ResidualBasedBlockBuilderAndSolver
...
             unsigned int k = row_begin;
-            for (auto it = indices[i].begin(); it != indices[i].end(); it++) {
+            for (auto it = indices[Index].begin(); it != indices[Index].end(); it++) {
                 Acol_indices[k] = *it;
                 Avalues[k] = 0.0;
                 k++;
             }
 
-            indices[i].clear(); //deallocating the memory
+            indices[Index].clear(); //deallocating the memory
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: @ashishdarekar I think you could leave the `i`, this would make it easier to review and also can prevent some mistakes

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 18]
```diff
@@ -1026,15 +1010,14 @@ class ResidualBasedBlockBuilderAndSolver
         const int ndofs = static_cast<int>(BaseType::mDofSet.size());
 
         // NOTE: dofs are assumed to be numbered consecutively in the BlockBuilderAndSolver
-        #pragma omp parallel for firstprivate(ndofs)
-        for (int k = 0; k<ndofs; k++) {
-            auto it_dof_iterator = it_dof_iterator_begin + k;
+        IndexPartition<std::size_t>(ndofs).for_each([&](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]:
```suggestion
        IndexPartition<std::size_t>(BaseType::mDofSet.size()).for_each([&](std::size_t Index){

```

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 19]
```diff
@@ -853,12 +844,10 @@ class ResidualBasedBlockBuilderAndSolver
         BaseType::mEquationSystemSize = BaseType::mDofSet.size();
         int ndofs = static_cast<int>(BaseType::mDofSet.size());
 
-        #pragma omp parallel for firstprivate(ndofs)
-        for (int i = 0; i < static_cast<int>(ndofs); i++) {
-            typename DofsArrayType::iterator dof_iterator = BaseType::mDofSet.begin() + i;
-            dof_iterator->SetEquationId(i);
-
-        }
+        IndexPartition<std::size_t>(ndofs).for_each([&, this](std::size_t Index){
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]:
```suggestion
        IndexPartition<std::size_t>(BaseType::mDofSet.size())).for_each([&, this](std::size_t Index){

```

**Review by philbucher — COMMENTED** [review]
minor comments, overall looks ok!

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 20]
```diff
@@ -592,13 +592,9 @@ class ResidualBasedBlockBuilderAndSolver
...
         // converged configuration at the end of the previous step.
-        const auto it_dof_begin = BaseType::mDofSet.begin();
-        #pragma omp parallel for
-        for (int i = 0; i < static_cast<int>(BaseType::mDofSet.size()); ++i) {
-            auto it_dof = it_dof_begin + i;
-            //NOTE: this is initialzed to - the value of dx prediction
-            dx_prediction[it_dof->EquationId()] = -(it_dof->GetSolutionStepValue() - it_dof->GetSolutionStepValue(1));
-        }
+        block_for_each(BaseType::mDofSet, [&](Dof<double>& rDof){
+            dx_prediction[rDof.EquationId()] = -(rDof.GetSolutionStepValue() - rDof.GetSolutionStepValue(1));
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h:597]:
```suggestion
            // NOTE: this is initialzed to - the value of dx prediction
            dx_prediction[rDof.EquationId()] = -(rDof.GetSolutionStepValue() - rDof.GetSolutionStepValue(1));

```

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 21]
```diff
@@ -988,16 +978,11 @@ class ResidualBasedBlockBuilderAndSolver
...
         //NOTE: dofs are assumed to be numbered consecutively in the BlockBuilderAndSolver
-        #pragma omp parallel for firstprivate(ndofs)
-        for (int k = 0; k<ndofs; k++) {
-            typename DofsArrayType::iterator dof_iterator = BaseType::mDofSet.begin() + k;
-
-            const int i = (dof_iterator)->EquationId();
-            (dof_iterator)->GetSolutionStepReactionValue() = -b[i];
-        }
+        block_for_each(BaseType::mDofSet, [&](Dof<double>& rDof){
+            const int i = rDof.EquationId();
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]:
```suggestion
            const std::size_t i = rDof.EquationId();

```

**Comment by philbucher** [comment]
@KratosMultiphysics/technical-committee this is ready, thx for the effort @ashishdarekar 

Since this is a very important class, could some of you also have a look to be safe?
I checked it now multiple times in detail but one never knows ...
@rubenzorrilla do you have time to also take a look?

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 22]
```diff
@@ -1564,78 +1539,68 @@ class ResidualBasedBlockBuilderAndSolver
...
-        #pragma omp parallel for firstprivate(nelements, ids)
-        for (int iii=0; iii<nelements; iii++) {
-            typename ElementsContainerType::iterator i_element = el_begin + iii;
-            pScheme->EquationId(*i_element, ids, CurrentProcessInfo);
-            for (std::size_t i = 0; i < ids.size(); i++) {
-                lock_array[ids[i]].lock();
-                auto& row_indices = indices[ids[i]];
-                row_indices.insert(ids.begin(), ids.end());
-                lock_array[ids[i]].unlock();
+        block_for_each(rModelPart.Elements(), ids, [&](Element& rElem, Element::EquationIdVectorType& rIds){
```
Comment (rubenzorrilla) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]:
```suggestion
        block_for_each(rModelPart.Elements(), ids, [&](Element& rElem, Element::EquationIdVectorType& rIdsTLS){

```
Minor suggestion on the variable naming. This is a completely personal opinion, but I like to name it in that way to make clear which containers are TLS and which one are captured from outside the lambda (in this case `rIds` can be easily misunderstood to be something passed by argument to the method when browsing the code).

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 23]
```diff
@@ -1564,78 +1539,68 @@ class ResidualBasedBlockBuilderAndSolver
...
-            typename ConditionsArrayType::iterator i_condition = cond_begin + iii;
-            pScheme->EquationId(*i_condition, ids, CurrentProcessInfo);
-            for (std::size_t i = 0; i < ids.size(); i++) {
-                lock_array[ids[i]].lock();
-                auto& row_indices = indices[ids[i]];
-                row_indices.insert(ids.begin(), ids.end());
-                lock_array[ids[i]].unlock();
+        });
+
+        block_for_each(rModelPart.Conditions(), ids, [&](Condition& rCond, Element::EquationIdVectorType& rIds){
```
Comment (rubenzorrilla) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: (same minor comment)

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 24]
```diff
@@ -1564,78 +1539,68 @@ class ResidualBasedBlockBuilderAndSolver
...
-                    row_indices.insert(slave_ids[i]);
-                    lock_array[slave_ids[i]].unlock();
+            struct TLS
+            {
+                Element::EquationIdVectorType master_ids;
+                Element::EquationIdVectorType slave_ids;
+            };
+            TLS tls;
+            tls.slave_ids.resize(3,0);
+            tls.master_ids.resize(3,0);
```
Comment (rubenzorrilla) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]:
```suggestion
            struct TLS
            {
                Element::EquationIdVectorType master_ids = Element::EquationIdVectorType(3,0);
                Element::EquationIdVectorType slave_ids = Element::EquationIdVectorType(3,0);
            };
            TLS tls;
```
Not 100% sure but I think this should work.

**Review by rubenzorrilla — COMMENTED** [review]
Changes make sense to me (only a few very few minor suggestions). For me this is ready to go in. I let @philbucher to give the final approval. Thanks!!

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h (hunk 22 above)
Comment (ashishdarekar) [reply kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: make sense !! Changed now.

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h (hunk 24 above)
Comment (ashishdarekar) [reply kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: Worked!!! Thanks @rubenzorrilla

**Review by rubenzorrilla — APPROVED** [review]

**Comment by rubenzorrilla** [comment]
Thanks @ashishdarekar !!

**Comment by philbucher** [comment]
@ashishdarekar I am running again the nightly tests manually here: https://github.com/KratosMultiphysics/Kratos/actions/runs/1115974716

Once they passed then you can merge this PR
Check the output of last nights nightly build to see which tests already fail on master

**Comment by philbucher** [comment]
seems to work, no new tests are failing
